pocket_ic/pocket_ic_server.py

```diff
@@ -109,44 +109,38 @@ def set_blob_store_entry(self, blob: bytes, compression: Optional[str]) -> str:
         """
         url = f"{self.url}/blobstore"
         if compression is None:
-            response = self.request_client.post(url, data=blob, headers=HEADERS)
+            response = self.request_client.post(url, data=blob)
         elif compression == "gzip":
-            headers = HEADERS | {"Content-Encoding": "gzip"}
+            headers = {"Content-Encoding": "gzip"}
             response = self.request_client.post(url, data=blob, headers=headers)
         else:
             raise ValueError('only "gzip" compression is supported')
 
         self._check_status_code(response)
         return response.text
 
-    def _get_url(self, pid: int) -> str:
-        tmp_dir = gettempdir()
-        ready_file_path = f"{tmp_dir}/pocket_ic_{pid}.ready"
-        port_file_path = f"{tmp_dir}/pocket_ic_{pid}.port"
-
+    def _get_url(self, port_file_path: int) -> str:
         stop_at = time.time() + 10  # Wait for the ready file for 10 seconds
 
-        while not os.path.exists(ready_file_path):
-            if time.time() < stop_at:
-                time.sleep(0.1)  # 100ms
-            else:
-                raise TimeoutError("PocketIC failed to start")
+        while True:
+            if os.path.isfile(port_file_path):
+                with open(port_file_path, "r", encoding="utf-8") as port_file:
+                    port = port_file.readline().strip()
+                if port:
```

Comment:
the condition should be that the port file contains a line terminated by a newline character

Reply:
Updated